### PR TITLE
Add rest areas to the toilet availability quest

### LINKED ISSUE #940: Rest area and highway=services have toilettes?
### General
Affected tag(s) to be modified/added: [](https://wiki.openstreetmap.org/wiki/Key:toilets)
Question asked: **Is there a toilet on the rest area?** yes/no

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

Perhaps we want to exclude areas where a toilet is mapped explict inside the area.

### PR DESCRIPTION
**Pull request by ENT8R**
Fixes #940

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -515,5 +515,6 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
     <string name="quest_construction_generic_title">Is this way completed?</string>
     <string name="quest_construction_building_title">Is this building completed?</string>
     <string name="questList_disabled_in_country">Never shown in %s</string>
+    <string name="quest_toiletAvailability_rest_area_title">Is there a toilet on this rest area?</string>
```
Comment (rugk) [inline app/src/main/res/values/strings.xml]: Is "on" really the right preposition here?

File: app/src/main/java/de/westnordost/streetcomplete/quests/toilet_availability/AddToiletAvailability.java [hunk 2]
```diff
@@ -20,9 +20,9 @@
 
 	@Override protected String getTagFilters()
 	{
-		// only for malls and big stores because users should not need to go inside a non-public
+		// only for malls, big stores and rest areas because users should not need to go inside a non-public
 		// place to solve the quest. (Considering malls and department stores public enough)
-		return "nodes, ways with shop ~ mall|department_store and name and !toilets";
+		return "nodes, ways with (shop ~ mall|department_store and name and !toilets) or (highway ~ services|rest_area and !toilets)";
```
Comment (rugk) [inline app/src/main/java/de/westnordost/streetcomplete/quests/toilet_availability/AddToiletAvailability.java]: BTW the query can be optimized if you put the "and not toilets" outside of everything else. It applies to both things, of course.

**Comment by westnordost** [comment]
You can make a new PR, this one has already been merged 

Am 19. April 2018 02:24:06 MESZ schrieb Richard Finegold <notifications@github.com>:
>goldfndr commented on this pull request.
>
>
>
>> @@ -515,5 +515,6 @@ Otherwise, you can download another keyboard in
>the app store. Popular keyboards
><string name="quest_construction_generic_title">Is this way
>completed?</string>
><string name="quest_construction_building_title">Is this building
>completed?</string>
><string name="questList_disabled_in_country">Never shown in %s</string>
>+    <string name="quest_toiletAvailability_rest_area_title">Is there a
>toilet on this rest area?</string>
>
>I suggest "Does this rest area have a toilet?"

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (rugk) [reply app/src/main/res/values/strings.xml]: Done in https://github.com/westnordost/StreetComplete/pull/1028

**Pull request merged**